### PR TITLE
Add PostgresAppTestExtension

### LINKED ISSUE #57: Create PostgresAppTestExtension
A dropwizard Application extension that also spins up an embedded Postgres database with migrations

### PR DESCRIPTION
**Pull request by chrisrohr**
Fixes #57

### CODE REVIEW

File: src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import io.zonky.test.db.postgres.embedded.LiquibasePreparer;
+import io.zonky.test.db.postgres.junit5.EmbeddedPostgresExtension;
+import io.zonky.test.db.postgres.junit5.PreparedDbExtension;
+import lombok.Getter;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * Multi-purpose extension for {@link Application} testing requiring a postgres database. The extension will spin up an
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java:17]: Probably ought to mention that we are assuming Liqiubase migrations here. That embedded Postgres actually supports both Liquibase and Flyway. Don't think we need to support both here, but should maybe mention it.

File: src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * embedded postgres database instance, run the migrations and then configure the {@link DropwizardAppExtension} with the
+ * database information.
+ * <p>
+ * To include this extension in an AppTest then add the following at the top of the class:
+ * <p>
+ * {@code @RegisterExtension}
+ * <br>
+ * {@code public static PostgresAppTestExtension<AppConfiguration> APP = new PostgresAppTestExtension("migrations.xml", "config.yml", App.class);}
+ * <p>
+ * The test instance will have access to the application and the postgres instance by calling {@code APP.getApp()} and {@code APP.getPostgres()}
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java:27]: Probably ought to capitalize all `postgres` as `Postgres` right? Or should we be formal and call it "PostgreSQL" which is the real name?

File: src/test/resources/PostgresAppTestExtensionTest/test-config.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+server:
+  type: simple
+  connector:
+    type: http
+    port: 10000
+
+database:
+  driverClass: org.postgresql.Driver
```
Comment (sleberknight) [inline src/test/resources/PostgresAppTestExtensionTest/test-config.yml:8]: I guess might as well make GitHub happy and add the newline here, and also for the migrations file.

File: src/test/resources/PostgresAppTestExtensionTest/test-migrations.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+</databaseChangeLog>
```
Comment (sleberknight) [inline src/test/resources/PostgresAppTestExtensionTest/test-migrations.xml:5]: ditto newline

File: src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java (hunk 1 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java]: Done

File: src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java (hunk 2 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java]: Done

File: src/test/resources/PostgresAppTestExtensionTest/test-config.yml (hunk 3 above)
Comment (chrisrohr) [reply src/test/resources/PostgresAppTestExtensionTest/test-config.yml]: Done

File: src/test/resources/PostgresAppTestExtensionTest/test-migrations.xml (hunk 4 above)
Comment (chrisrohr) [reply src/test/resources/PostgresAppTestExtensionTest/test-migrations.xml]: Done

File: src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import lombok.Getter;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/**
+ * Multi-purpose extension for {@link Application} testing requiring a PostgreSQL database. The extension will spin up an
+ * embedded PostgreSQL database instance, run the migrations and then configure the {@link DropwizardAppExtension} with the
+ * database information.
+ * @implNote The embedded PostgreSQL extension supports both Flyway and Liquibase, but we are assuming migrations are Liquibase.
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/test/dropwizard/app/PostgresAppTestExtension.java:20]: I think `@implNote` is always supposed to be grouped with the other `@xxx` at bottom. Suggest just putting this as a separate paragraph.

**Pull request merged**